src/crimson/osd/pg_recovery.cc

```diff
@@ -514,6 +520,23 @@ void PGRecovery::request_primary_scan(
   });
 }
 
+PGRecovery::interruptible_future<>
+PGRecovery::recover_object_with_throttle(
+  const hobject_t &soid,
+  eversion_t need)
+{
```

Comment:
I think we can push a log here to capture the details about ongoing operation so that it would be easier to analyze logs in case of throttle vs non throttle.

Reply:
Yeah, will do.